### DISCUSSION TITLE
Update Automad shows error message

### DISCUSSION THREAD

**Discussion by derkolarsky** [automadcms/discussions#27] [Q&A]
Hey everyone,

Maybe just a little issue that I ran into: when I go to system settings and click the update button I get this error message:
**Can't connect to the update server! This could typically be caused by a wrong update configuration (config/config.php), a SSL certificate error or a missing cURL library.**
The server has a certificate from Let's Encrypt set up via certbot on Ubuntu.
curl is installed (8.5.0-2ubuntu10.4) as well.

Strange thing: installing packages works fine. Wouldn't that normally also show some error then?

The content of the config.php is – I guess – the normal setup after installation:
```
<?php return <<< JSON
{
    "AM_ALLOWED_FILE_TYPES": "dmg, iso, rar, tar, zip, aiff, m4a, mp3, ogg, wav, ai, dxf, eps, gif, ico, jpg, jpeg, png, psd, svg, tga, tiff, avi, flv, mov, mp>
    "AM_CACHE_ENABLED": true,
    "AM_CACHE_LIFETIME": 43200,
    "AM_CACHE_MONITOR_DELAY": 120,
    "AM_DEBUG_ENABLED": false,
    "AM_FEED_ENABLED": true,
    "AM_FEED_FIELDS": "+hero, +main",
    "AM_FILE_GUI_TRANSLATION": "",
    "AM_HEADLESS_ENABLED": false
}
JSON;
```

Where might I have missed to define something for the connection to work…? 🤔 

Thanks in advance
Ralf

**Comment by marcantondahmen** [comment]
Hi there! I guess your server is missing a root certificate in order to be able to use cURL with SSL. This should normally be installed on common web servers out of the box. 

The Let' Encrypt certificate is only used for secure connections from a client to your domain. But in order to update your installation, your server becomes the client that tries to establish a secure connection to the update server.

**Reply by derkolarsky** [reply]
Okay. Thanks for your response.

Since I'm experimenting with my local server at home that can be reached via DDNS forward from my hoster, I might not solve this situation soon. But that's okay for now… I'll dive deeper in the Ubuntu linux world and maybe I'll find a solution for this.

**Reply by derkolarsky** [reply]
Found out that console mode with `php automad/console update` doesn't show any error and that I'm up-to-date.
Could be interesting to see if on an upcoming update I'll have the chance to update Automad this way… 🤷

**Reply by marcantondahmen** [reply]
Hi, yeah I understand that tinkering with a self-build server is great fun and quite enjoyable, even when this means to sometimes make your hands dirty ...

Unfortunately the console update will probably also fail during the update when the root certificate can't be found. 

One important thing is that Automad 1.10.9 is the latest v1 version so far. Since Automad v2 ships some breaking changes, it is not possible to update that way from v1 to v2. Sites have to be migrated from v1 to v2. And this process has limitations.
